jenkins/pipelines/java/Jenkinsfile

```diff
@@ -17,17 +16,24 @@ pipeline {
                 }
             }
         }
-        stage('Run Tests') {
+
+        // Note that there is no fundamental reason that the Desktop and Web Services
+        // builds couldn't run in parallel.  We'd need to figure out how to set
+        // up the environment server and sg at alternate ports, probably, and we'd
+        // need hardware that can support the load.
+
+        stage('Run Desktop Tests') {
             parallel {
                 stage('OSX Tests') {
                     agent { label 'mob-e2e-mac-01' }
+                    options {
+                        timeout(time: 45, unit: 'MINUTES')
+                        lock("mob-e2e-mac-01")
```

Comment:
This is not going to work unless you've set up a lockable resource on Jenkins.  It's probably confusing but I named them the same as their device name to make it clear what they are.